### PR TITLE
P#62389 #2781085 Address list without of a name created

### PR DESCRIPTION
**Pull request by vu-vuong-esg**
P#62389 #2781085 Address list without of a name created
#161

### CODE REVIEW

**Comment by jayay** [comment]
I don't quite understand the change?

**Comment by vu-vuong-esg** [comment]
> I don't quite understand the change?

i update it, can you check PR ?

File: plugin/Record/RecordManagerUpdateListViewAddress.php [hunk 1]
```diff
@@ -94,7 +94,10 @@ private function insertNewRecords(array $records, string $table): bool
 		$pWpDb = $this->getWpdb();
 		$result = true;
 		foreach ($records as $record) {
-			$result = $result && $pWpDb->insert($prefix.$table, $record);
+			if(is_array($record))
```
Comment (jayay) [inline plugin/Record/RecordManagerUpdateListViewAddress.php:97]: This is better. @vu-vuong-esg do you know, under which circumstances `$record` is not an array?

Comment (vu-vuong-esg) [reply plugin/Record/RecordManagerUpdateListViewAddress.php:97]: because $record is string so it doesn't know column in table insert.

**Comment by vu-vuong-esg** [comment]
as in the chat, i add code in fixed error create address list can't save page.

File: plugin/Record/RecordManagerUpdateListViewAddress.php (hunk 1 above)
Comment (jayay) [reply plugin/Record/RecordManagerUpdateListViewAddress.php:97]: In which case is `$record` a string? What is the root cause?

**Comment by tang-hien-egs** [comment]
@jayay , i fixed it. can you check PR ?

**Comment by jayay** [comment]
Isn't the origin of the bug in the way the JavaScript code submits the data to be saved?

**Comment by vu-vuong-esg** [comment]
> Isn't the origin of the bug in the way the JavaScript code submits the data to be saved?

@jayay , No, JavaScript code submits data true

File: plugin/Record/RecordManagerUpdateListViewAddress.php (hunk 1 above)
Comment (jayay) [reply plugin/Record/RecordManagerUpdateListViewAddress.php:97]: @vu-vuong-esg are you sure `$row` has the correct type here and doesn't need to be wrapped in another array?
https://github.com/onOfficeGmbH/oo-wp-plugin/blob/f328ccafd4dc37a672ab9d33b156d00e071ac263/plugin%2FGui%2FAdminPageAddressListSettings.php#L233-L236

**Review by jayay — APPROVED** [review]

**Comment by michaelarnold-dev** [comment]
@tang-hien-egs Could you try to increase the code coverage?

**Review by jayay — APPROVED** [review]

**Review by jmaas-onoffice — APPROVED** [review]

**Review by jmaas-onoffice — CHANGES_REQUESTED** [review]
I can still reproduce the error with the [latest code](https://github.com/onOfficeGmbH/oo-wp-plugin/pull/172#issuecomment-1094942595).

I have configured this version at https://3.wptest.onofficeweb.com/. If you create a new address list there without entering a name and save, then the list will actually still be created without a name.

Could you check this on the test instance please?

**Comment by tang-hien-egs** [comment]
> I can still reproduce the error with the [latest code](https://github.com/onOfficeGmbH/oo-wp-plugin/pull/172#issuecomment-1094942595).
> 
> I have configured this version at https://3.wptest.onofficeweb.com/. If you create a new address list there without entering a name and save, then the list will actually still be created without a name.
> 
> Could you check this on the test instance please?
@jmaas-onoffice 
this errors like link record : https://files.fm/f/gzgshnbxc , your mean is when created address list, we must enter name address not exist

**Comment by jmaas-onoffice** [comment]
> this errors like link record : https://files.fm/f/gzgshnbxc , your mean is when created address list, we must enter name address not exist  ?

Yes, the video shows the error. With estate lists and forms, when you try to save without a name, it gives an error and you can't save. The same should happen for address lists.

Is the problem clear? :)

**Comment by tang-hien-egs** [comment]
> > this errors like link record : https://files.fm/f/gzgshnbxc , your mean is when created address list, we must enter name address not exist  ?
> 
> Yes, the video shows the error. With estate lists and forms, when you try to save without a name, it gives an error and you can't save. The same should happen for address lists.
> 
> Is the problem clear? :)

yes, i will fixed it as soon as !

**Comment by tang-hien-egs** [comment]
@jmaas-onoffice I have a issue I don't know if it is in your request or not ? This is, With estate lists, forms and address list, when you enter name new and you press "Save" 2x in a row, it created data success. See video in link : https://files.fm/f/q7c5wyrxu

**Comment by jmaas-onoffice** [comment]
@tang-hien-egs Oh, thanks for reporting. That is a bit weird. But since the last changes will be saved, I think this is ok. We can fix that later.

**Comment by vu-vuong-esg** [comment]
> @tang-hien-egs Oh, thanks for reporting. That is a bit weird. But since the last changes will be saved, I think this is ok. We can fix that later.
> 
> Is the code ready for review apart for this issue?

@jmaas-onoffice  yes, the code ready for review.

**Review by jmaas-onoffice — APPROVED** [review]

**Review by jmaas-onoffice — APPROVED** [review]

**Pull request merged**